### PR TITLE
Search

### PR DESCRIPTION
**Pull request by IC-Socaciu**
This is an intermediary phase of my US4 regarding the search functionality. At this moment I am receiving an error message which will be shared on the Question board.

### CODE REVIEW

**Comment by IC-Socaciu** [comment]
For now the search works with the included tags, ex: **Undertaker**
Later in the stages of the project I will also add the lower case and multiple words. 
However, for US4 this was the intended stage of the search functionality.

File: components/Search/SearchResults/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { interviews } from "@/public/interviews";
+
+export default function SearchResults({ searchTerm }) {
+  const results = interviews.filter((interview) =>
+    interview.tags.some((tag) => tag.includes(searchTerm))
```
Comment (CJAkkuec) [inline components/Search/SearchResults/index.js:5]:
```suggestion
        interview.tags.some((tag) => tag.toLowerCase().includes(searchTerm.toLowerCase()))
```
If you sanitize the tags and the searchTerm where you check one against the other, it won't matter whether the original inputs/etc were uppercase or lowercase. :)

File: components/Search/SearchResults/index.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        <li key={interview.id}>
+          <h3>{interview.title}</h3>
+          <p>{interview.text}</p>
+          <ul>
+            {interview.tags.map((tag) => (
+              <li key={tag}>{tag}</li>
+            ))}
+          </ul>
+        </li>
+      ))}
```
Comment (CJAkkuec) [inline components/Search/SearchResults/index.js:24]: I'm not sure if you need to offer the entire article as a search result. A link to the article page would probably be more than enough. If you want to include a preview of the article text, you might want to add some overflow rules, such as

```js
//...

<PreviewText>{interview.text}</PreviewText>

//...

const PreviewText = styled.p`
text-overflow: ellipsis;
white-space: nowrap;
overflow: hidden;
`
```
Maybe you'd also not want to include the raw markdown text?

File: public/interviews.js [hunk 3]
```diff
@@ -344,3 +351,31 @@ As always, a huge *Thank You* to the band for their time, openness and availabil
...
+          <ul>
+            {interview.tags.map((tag) => (
+              <li key={tag}>{tag}</li>
+            ))}
+          </ul>
+        </li>
+      ))}
+    </ul>
+  );
+}
```
Comment (CJAkkuec) [inline public/interviews.js:381]: Probably doesn't need to be part of the PR. 👀 ?

**Review by CJAkkuec — CHANGES_REQUESTED** [review]
Hey there! Looking good, just a few notes. 👍 

... and yes, handling a search feature isn't that easy.

File: components/InterviewCard/index.js [hunk 4]
```diff
@@ -1,18 +1,17 @@
...
-        <ImageContainer>
-          <Image src={image} alt={title} />
-        </ImageContainer>
-        <SummaryContainer>{summary}</SummaryContainer>
-        <ReadLink href={`/post/${id}`}>Read more</ReadLink>
-        <SeparatorLine />
-      </CardContainer>
-    </>
+    <CardContainer>
+      <TitleContainer>{title}</TitleContainer>
```
[the remark this answers is not in the document]
Comment (IC-Socaciu) [reply components/InterviewCard/index.js:7]: Done :)

File: components/Search/SearchInput/index.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        />
+        <StyledSearchButton type="submit">
+          <span>🔍</span> Search
+        </StyledSearchButton>
+      </form>
+      <h3>
+        {searchTerm
+          ? `You searched for "${searchTerm}"`
+          : "Please search for a keyword"}
+      </h3>
```
[the remark this answers is not in the document]
Comment (IC-Socaciu) [reply components/Search/SearchInput/index.js:28]: Removed for now. :)

File: components/Search/SearchResults/index.js (hunk 1 above)
Comment (IC-Socaciu) [reply components/Search/SearchResults/index.js:5]: Works like a charm! Thank you.

File: components/Search/SearchResults/index.js (hunk 2 above)
Comment (IC-Socaciu) [reply components/Search/SearchResults/index.js:24]: Looks better. I tried the preview for now and maybe make a new US and focus on refining the search. Will have to sketch out something during the weekend.

File: public/interviews.js (hunk 3 above)
Comment (IC-Socaciu) [reply public/interviews.js:381]: I most likely created a snafu and will have to pay more attention XD

**Review by julialoeschel — COMMENTED** [review]
LGTM, 
just found some code that is not used anymore (I think) 

🐜

File: public/interviews.js [hunk 6]
```diff
@@ -344,3 +351,31 @@ As always, a huge *Thank You* to the band for their time, openness and availabil
     liked: false,
   },
 ];
+
+// SEARCH FUNCTION TEST
+
+export default function SearchResults({ searchTerm }) {
```
[the remark this answers is not in the document]
Comment (IC-Socaciu) [reply public/interviews.js]: Overexcited that it works and totally forgot! All gone! :D

**Pull request merged**